serialized_data_interface/__init__.py

```diff
@@ -125,14 +151,28 @@ def get_data(self) -> list:
 
         return data
 
-    def send_data(self, data: dict):
-        data = yaml.dump(data)
-        for rel in self._relations:
+    def send_data(self, data: dict, app_name: str = None):
+        serialized = yaml.dump(data)
+
+        if len(set(self.versions.values())) != 1 and app_name is None:
+            raise AppNameOmitted(self.relation_name, self.versions)
+
+        relations = self._relations
+
+        if app_name is not None:
+            relations = [r for r in relations if r.app.name == app_name]
+
+        if not relations:
+            raise InvalidAppName(self.relation_name, app_name)
```

Comment:
I'm not 100% sure what should happen here, but if len(versions)==1 and app_name is None then the `InvalidAppName` will trigger regardless of the relation.  So then I think line 157 could test `len(...) != 1 or app_name is None`?

Reply:
@ca-scribner added comments that explain it a bit more, do they make sense?